scripts/artifacts-building/user_rcbops_artifacts_building.yml

```diff
@@ -31,3 +31,30 @@ openstack_repo_url: "{{ rpco_mirror_base_url }}"
 # default variant.
 lxc_image_cache_server: images.linuxcontainers.org
 
+# When we build python/container artifacts we pull the default
+# LXC cache variant from LXC upstream which may have packages
+# installed which are more recent than those we have available
+# in our artifacted apt repository. Here we figure out which
+# packages are installed, but not available in a configured
+# source, then we downgrade them to the latest available in
+# the configured sources.
+lxc_cache_prep_pre_commands: |
+    if [ -a /etc/resolv.conf ]; then
+      mv /etc/resolv.conf /etc/resolv.conf.org
+    fi
+    {% for resolver in lxc_cache_prep_dns %}
+    echo "nameserver {{ resolver }}" >> /etc/resolv.conf
+    {% endfor %}
+    apt-key add /root/repo.keys
+    export DEBIAN_FRONTEND=noninteractive
+    apt-get update
+    if apt list --installed 2>/dev/null | grep '\/now'; then
+      pkg_downgrade_list=$(apt list --installed 2>/dev/null | awk -F / '/\/now/ {print $1}')
+      pkg_downgrade_list_versioned=$(for pkg in ${pkg_downgrade_list}; do echo -n "${pkg}=$(apt-cache madison ${pkg} | head -n 1 | awk '{ print $3 }') "; done)
+      apt-get install -y --force-yes ${pkg_downgrade_list_versioned}
```

Comment:
While I think it's a generally good idea to force what we are using, this forces packages (some packages that we don't need) to be marked as explicitly installed (and downgraded) with theirs deps. 

Because I don't know if we are opening pandora's box, I'd rather let someone else (with more apt experience than me) review this.